src/bitbucket/__tests__/BitbucketPipelinesAPI.test.ts

```diff
@@ -0,0 +1,37 @@
+import axios from 'axios';
+import { BitbucketPipelinesAPI } from '../BitbucketPipelinesAPI';
+
+jest.mock('axios');
+const mockedAxios = axios as unknown as jest.Mocked<typeof axios>;
+
+jest.mock('../BitbucketAuthenticator', () => ({
+  bitbucketAuthenticator: {
+    getAuthConfig: () => Promise.resolve({}),
+  },
+}));
+
+const bitbucketPipelineAPI = new BitbucketPipelinesAPI({
+  repoName: 'repo',
+  repoOwner: 'owner',
+});
+
+describe('BitbucketPipelinesAPI', () => {
```

Comment:
We should add a `beforeEach` that clears/resets mocks to ensure they don't leak between tests. Saves having to refactor it when we add a test for next time.

You might have to re-apply the `BitbucketAuthenticator` mock in the beforeEach if you reset the mocks. Otherwise you'll need to update ` mockedAxios.get.mockResolvedValue(response)` to only mock it once or manually reset that mock so it doesn't persist across tests.

Reply:
Yup, I have rested the mocks in `beforeEach` and moved the mocks for `BitbucketAuthenticator ` within `beforeEach`